### PR TITLE
Initialize PublicApp only if the public API flag is set

### PR DESCRIPTION
**Pull request by tanghel**
## Type
- [ ] Bug
- [x] Feature  
- [ ] Refactoring
- [ ] Performance improvement

## Proposed Changes
- Initialize PublicApp only if the public API flag is set

## How to test
- Modules dependent from the public app should not be initialized unless the public api flag is set

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -27,93 +27,91 @@ import { PaginationInterceptor } from './interceptors/pagination.interceptor';
...
-      new CachingInterceptor(
-        cachingService,
-        httpAdapterHostService,
-        metricsService,
-        protocolService,
-      ),
+  if (apiConfigService.getIsPublicApiActive()) {
+    const publicApp = await NestFactory.create<NestExpressApplication>(PublicAppModule);
+    publicApp.use(bodyParser.json({ limit: '1mb' }));
+    publicApp.use(requestIp.mw());
```
Comment (danielailie) [inline src/main.ts:47]: maybe here you can extract a method configurePublicApi() or getPublicApi

File: src/main.ts [hunk 2]
```diff
@@ -27,93 +27,91 @@ import { PaginationInterceptor } from './interceptors/pagination.interceptor';
...
+    publicApp.enableCors();
+    publicApp.useLogger(publicApp.get(WINSTON_MODULE_NEST_PROVIDER));
+    publicApp.disable('etag');
+    publicApp.disable('x-powered-by');
+
```
Comment (danielailie) [inline src/main.ts:52]: move this closer to their usage and define them only if need it

**Review by claudiulataretu — DISMISSED** [review]

**Review by danielailie — APPROVED** [review]

**Review by gabrielmatei — DISMISSED** [review]

**Review by CatalinFaurPaul — DISMISSED** [review]

File: src/main.ts (hunk 1 above)
Comment (tanghel) [reply src/main.ts]: indeed

File: src/main.ts (hunk 2 above)
Comment (tanghel) [reply src/main.ts]: these are part of the default configuration, have no usage in main.ts

Comment (tanghel) [reply src/main.ts:47]: fixed

**Review by danielailie — APPROVED** [review]

**Pull request merged**